gef-ui/server/controllers/file/index.js

```diff
@@ -0,0 +1,30 @@
+const stream = require('stream');
+const { downloadFile } = require('../../services/api');
+
+const generateDownload = async (req, res) => {
+  const {
+    params: { fileId },
+    query: { filename },
+    session: { userToken },
+  } = req;
+
+  try {
+    const file = await downloadFile(fileId, userToken);
+
+    if (!file) {
+      res.status(404).send('Cannot file file');
+      return;
+    }
+
+    if (filename) res.set('Content-disposition', `attachment; filename=${filename}`);
+    res.set('Content-Type', file.headers['content-type']);
+
+    const readStream = new stream.PassThrough();
+    file.pipe(readStream).pipe(res);
+  } catch (err) {
+    console.log(`Error creating download for ${fileId}: ${err}`);
+    res.status(500);
```

Comment:
do we need a better error experience for this and 404 above?

Reply:
Given this is just a download 404 made the most sense

Reply:
if we are saying theres no ui to this then pls mark as resolved